src/src/rx-serial/SerialTramp.cpp

```diff
@@ -0,0 +1,113 @@
+#include "SerialTramp.h"
+#include "msptypes.h"
+#include "freqTable.h"
+
+// Calculate tramp protocol checksum of provided buffer
+uint8_t checksum(uint8_t *buf)
+{
+    uint8_t cksum = 0;
+
+    for (int i = 1 ; i < TRAMP_FRAME_SIZE - 2; i++) {
+        cksum += buf[i];
+    }
+
+    return cksum;
+}
+
+void SerialTramp::sendQueuedData(uint32_t maxBytesToSend)
+{
+    uint32_t bytesWritten = 0;
+    while (_fifo.size() > 0 && bytesWritten < maxBytesToSend) {
+        _fifo.lock();
+        uint8_t frameSize = _fifo.pop() - 1;
+        uint8_t frame[frameSize];
+        _fifo.popBytes(frame, frameSize);
```

Comment:
Should we be locking and unlocking the `_fifo` for all these read operations, so that we are using the compile flags to protect the queue during r/w?

Reply:
Oh we are lol. We do it higher up. Can we move the lock right be fore the popBytes so that this doesn't catch the next person out?

Reply:
Hmm nevermind. there's a rogue `pop` in there that also gets covered by the lock. Ignore me.